src/pcm/SoxrResampler.cxx

```diff
@@ -80,20 +88,110 @@ soxr_parse_quality(const char *quality) noexcept
 	return SOXR_INVALID_RECIPE;
 }
 
+static unsigned 
+soxr_parse_precision(unsigned value) {
+	switch(value) {
+		case 16:
+		case 20:
+		case 24:
+		case 28:
+		case 32:
+			break;
+		default:
+			throw FormatInvalidArgument("soxr converter invalid precision : %d [16|20|24|28|32]",
+					    value);
+			break;
+	}
+	return value;
+}
+
+static double
+soxr_parse_phase_respons(signed value) {
```

Comment:
Why make this `signed` if negative values are not allowed?
And why is this integer if it implicitly casts to floating point?